packages/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js

```diff
@@ -0,0 +1,896 @@
+    }
+    const { type } = node;
+    return unmemoType(type) === unmemoType(matchingType);
+  }
+
+  elementToNode(element) {
+    return elementToTree(element);
+  }
+
+  nodeToHostNode(node, supportsArray = false) {
+    const nodes = nodeToHostNode(node);
+    if (Array.isArray(nodes) && !supportsArray) {
+      return nodes[0];
+    }
+    return nodes;
+  }
+
+  displayNameOfNode(node) {
+    if (!node) return null;
+    const { type, $$typeof } = node;
+
+    const nodeType = type || $$typeof;
+
+    // newer node types may be undefined, so only test if the nodeType exists
+    if (nodeType) {
+      switch (nodeType) {
+        case ConcurrentMode || NaN: return 'ConcurrentMode';
+        case Fragment || NaN: return 'Fragment';
+        case StrictMode || NaN: return 'StrictMode';
+        case Profiler || NaN: return 'Profiler';
+        case Portal || NaN: return 'Portal';
+        case Suspense || NaN: return 'Suspense';
+        default:
+      }
+    }
+
+    const $$typeofType = type && type.$$typeof;
+
+    switch ($$typeofType) {
+      case ContextConsumer || NaN: return 'ContextConsumer';
+      case ContextProvider || NaN: return 'ContextProvider';
+      case Memo || NaN: {
+        if (type.displayName) {
+          return type.displayName;
+        }
+        const name = this.displayNameOfNode({ type: type.type });
+        // "works on a memoized functional component" test desires `Memo()` instead of `Memo`
+        return `Memo(${name})`;
+      }
+      case ForwardRef || NaN: {
+        if (type.displayName) {
+          return type.displayName;
+        }
+        const name = displayNameOfNode({ type: type.render });
+        return name ? `ForwardRef(${name})` : 'ForwardRef';
+      }
+      case Lazy || NaN: {
+        return 'lazy';
+      }
+      default: return displayNameOfNode(node);
+    }
+  }
+
+  isValidElement(element) {
+    return isElement(element);
+  }
+
+  isValidElementType(object) {
+    return !!object && isValidElementType(object);
+  }
+
+  isFragment(fragment) {
+    return typeOfNode(fragment) === Fragment;
+  }
+
+  isCustomComponent(type) {
+    const fakeElement = makeFakeElement(type);
+    return !!type && (
+      typeof type === 'function'
+      || isForwardRef(fakeElement)
+      || isContextProvider(fakeElement)
+      || isContextConsumer(fakeElement)
+      || isSuspense(fakeElement)
+    );
+  }
+
+  isContextConsumer(type) {
+    return !!type && isContextConsumer(makeFakeElement(type));
+  }
+
+  isCustomComponentElement(inst) {
+    if (!inst || !this.isValidElement(inst)) {
+      return false;
+    }
+    return this.isCustomComponent(inst.type);
+  }
+
+  getProviderFromConsumer(Consumer) {
+    // React stores references to the Provider on a Consumer differently across versions.
+    if (Consumer) {
+      let Provider;
+      if (Consumer._context) {
+        ({ Provider } = Consumer._context);
+      }
+      if (Provider) {
+        return Provider;
+      }
+    }
```

Comment:
In React 17+, when is `Consumer._context` not set?

Reply:
Never. But then shallow() rendering isn't a thing anyway which relies on this. So either we drop shallow rendering and respect what's actually happening in React or we support shallow rendering and have these guards in place. I'll remove the comment which is no longer relevant.

Reply:
Never. But there's a test specifically for passing something other than `createContext` so these guards are required. I'll remove the outdated comment.